akka-persistence-testkit/src/main/scala/akka/persistence/testkit/internal/SerializedEventStorageImpl.scala

```diff
@@ -5,39 +5,43 @@
 package akka.persistence.testkit.internal
 
 import scala.util.Try
-
 import akka.actor.{ ActorSystem, ExtendedActorSystem }
 import akka.annotation.InternalApi
 import akka.persistence.PersistentRepr
 import akka.persistence.testkit.EventStorage
+import akka.persistence.testkit.EventStorage.Metadata
 import akka.serialization.{ Serialization, SerializationExtension }
 
 /**
  * INTERNAL API
+ * FIXME, once we add serializers for metadata serialize the metadata payload if present
  */
 @InternalApi
 private[testkit] class SerializedEventStorageImpl(system: ActorSystem) extends EventStorage {
-
-  override type InternalRepr = (Int, Array[Byte])
+  override type InternalRepr = (Int, Array[Byte], Metadata)
 
   private lazy val serialization = SerializationExtension(system)
 
   /**
    * @return (serializer id, serialized bytes)
    */
-  override def toInternal(repr: PersistentRepr): (Int, Array[Byte]) =
+  override def toInternal(repr: (PersistentRepr, Metadata)): (Int, Array[Byte], Metadata) =
```

Comment:
does it have to be tuple, or can we use two params?

Reply:
I'll see about making it a separate param

Reply:
maybe now the internal type is not a tuple it is okay as is